Sources/Sentry/SentrySystemEventBreadcrumbs.m

```diff
@@ -177,6 +181,58 @@ - (void)initScreenshotObserver
                                                  name:UIApplicationUserDidTakeScreenshotNotification
                                                object:nil];
 }
+
+- (NSNumber *_Nullable)storedTimezoneOffset
+{
+    SentryAppStateManager *appStateManager =
+        [SentryDependencyContainer sharedInstance].appStateManager;
+    SentryAppState *currentState = [appStateManager loadCurrentAppState];
+    return currentState.timezoneOffset;
+}
+
+- (void)initTimezoneObserver
+{
+    // Detect if the stored timezone is different from the current one;
+    // if so, then we also send a breadcrumb
+    NSNumber *_Nullable storedTimezoneOffset = [self storedTimezoneOffset];
+
+    if (storedTimezoneOffset == nil) {
+        [self updateStoredTimezone];
+    } else if (storedTimezoneOffset.doubleValue != [NSTimeZone localTimeZone].secondsFromGMT) {
+        [self timezoneEventTriggered];
+    }
+
+    // Posted when the timezone of the device changed
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(timezoneEventTriggered)
+                                                 name:NSSystemTimeZoneDidChangeNotification
+                                               object:nil];
+}
+
+- (void)timezoneEventTriggered
+{
```

Comment:
`m`: We could pass `storedTimezoneOffset` to `timezoneEventTriggered` to avoid one IO operation. We should also remove the below usage of `[self storedTimezoneOffset]`.
```suggestion
        [self timezoneEventTriggered:storedTimezoneOffset];
    }

    // Posted when the timezone of the device changed
    [[NSNotificationCenter defaultCenter] addObserver:self
                                             selector:@selector(timezoneEventTriggered)
                                                 name:NSSystemTimeZoneDidChangeNotification
                                               object:nil];
}

- (void)timezoneEventTriggered {
    [self timezoneEventTriggered:nil];
}

- (void)timezoneEventTriggered:(NSNumber *)storedTimezoneOffset
{
    if (storedTimezoneOffset == nil) {
        storedTimezoneOffset = [self storedTimezoneOffset];
    }
```

Reply:
Personally I feel the code got significant more complex just to save one read operation when the timezone changed while the app wasn't open, but fair enough :)